ruby/ql/lib/codeql/ruby/security/SqlInjectionCustomizations.qll

```diff
@@ -51,6 +52,23 @@ module SqlInjection {
    * sanitizer-guard.
    */
   class StringConstArrayInclusionCallAsSanitizer extends Sanitizer,
-    StringConstArrayInclusionCallBarrier
-  { }
+    StringConstArrayInclusionCallBarrier { }
+
+  /**
+   * A call to `Mysql2::Client.escape`, considered as a sanitizer.
+   */
+  class Mysql2EscapeSanitization extends Sanitizer {
+    Mysql2EscapeSanitization() {
+      this = API::getTopLevelMember("Mysql2").getMember("Client").getAMethodCall("escape")
+    }
+  }
+
+  /**
+   * A call to `SQLite3::Database.quote`, considered as a sanitizer.
+   */
+  class SQLite3EscapeSanitization extends Sanitizer {
+    SQLite3EscapeSanitization() {
+      this = API::getTopLevelMember("SQLite3").getMember("Database").getAMethodCall("quote")
+    }
+  }
```

Comment:
On a separate note, I think that these are correct but don't technically function as sanitizers at the moment because we lack flow summaries for `escape` and `quote` - so there's no dataflow through them to sanitize.

For SQLite3 at least, the [`quote`](https://www.rubydoc.info/github/luislavena/sqlite3-ruby/SQLite3%2FDatabase.quote) method doesn't modify the input string but returns the modified version of the string. We should have a flow summary for this method (non-value-preserving flow from `Argument[0]` to `ReturnValue`) to account for this. I think that the same is true for `quote` in MySQL3.